### DISCUSSION TITLE
How to use Symfony Web Profiler toolbar from a separate container?

### DISCUSSION THREAD

**Discussion by mindaugasw** [symfony/symfony#49366] [Q&A]
My application is split into 2 Docker containers: Frontend app (only static content, HTML, CSS, JS) and backend API (Symfony application). So the browser loads frontend app (`http://localhost:8030`) which then makes requests to the API on a different host/container (`http://localhost:8031`).

How can I use Symfony Web Profiler toolbar with this setup?

Normally Symfony injects HTML for the toolbar into each returned HTML response. But in my case Symfony never returns any HTML and acts only as the API for Json data.
I can see all requests in the profiler by going to `http://localhost:8031/_profiler/search`, but I'd like to have the full toolbar in the frontend app loaded through `http://localhost:8030`.

Is it possible to somehow manually inject profiler HTML code into responses returned _not_ by Symfony application?

**Comment by MatTheCat** [comment]
The toolbar is linked to a request handled by Symfony so it would make no sense to include it in another context (what if your frontend never calls your backend? Or many times?).

What you could do though is to collect the `X-Debug-Token-Link` header of Symfony’s responses and display them one way or another.

https://symfony.com/blog/new-in-symfony-2-4-quicker-access-to-the-profiler-when-working-on-an-api

**Reply by mindaugasw** [reply]
Expected behaviour would be that initially toolbar would be empty (without initial request for HTML, because it was served not through Symfony) and would only show info about AJAX requests when they're made.

Thanks for telling about headers, I'll try to implement some similar AJAX requests table myself.
